task_sdk/tests/api/test_client.py

```diff
@@ -82,6 +83,122 @@ def handle_request(request: httpx.Request) -> httpx.Response:
         assert err.value.args == ("Not found",)
         assert err.value.detail is None
 
+    @mock.patch("time.sleep", return_value=None)
+    def test_retry_handling_unrecoverable_error(self, mock_sleep):
+        responses: list[httpx.Response] = [
+            *[httpx.Response(500, text="Internal Server Error")] * 11,
+            httpx.Response(200, json={"detail": "Recovered from error - but will fail before"}),
+            httpx.Response(400, json={"detail": "Should not get here"}),
+        ]
+
+        def mock_handle_request(request: httpx.Request) -> httpx.Response:
+            return responses.pop(0)
+
+        client = Client(
+            base_url=None,
+            dry_run=True,
+            token="",
+            mounts={"'http://": httpx.MockTransport(mock_handle_request)},
+        )
+
```

Comment:
The section of code has been duplicated multiple times. Should we consider getting them out?

```
@pytest.fixture
def mock_client():
    """Helper fixture to create a mock client with custom responses."""
    def _create_client(responses):
        def mock_handle_request(request: httpx.Request) -> httpx.Response:
            return responses.pop(0)
        return Client(
            base_url=None,
            dry_run=True,
            token="",
            mounts={"'http://": httpx.MockTransport(mock_handle_request)},
        )
    return _create_client
```

and use `client = mock_client(responses)`

Reply:
Good point. Thanks for the proposal. Consolidated now... but w/o fixture as this does not provide a benefit over a "normal" utility like it was there before.